ReactiveCocoa/Swift/Atomic.swift

```diff
@@ -8,9 +8,56 @@
 
 import Foundation
 
+public protocol MutexType {
+	func lock()
+	func unlock()
+}
+
+public final class RecursiveLock: MutexType {
+	private var _lock: NSRecursiveLock
+
+	public init(_ name: String) {
+		_lock = NSRecursiveLock()
+		_lock.name = name
+	}
+
+	public func lock() {
+		_lock.lock()
+	}
+
+	public func unlock() {
+		_lock.unlock()
+	}
+}
+
+public final class PosixThreadMutex: MutexType {
+	private var mutex = pthread_mutex_t()
+	
+	public init() {
+		let result = pthread_mutex_init(&mutex, nil)
+		assert(result == 0, "Failed to initialize mutex with error \(result).")
```

Comment:
These should be `precondition`s, I don't think we want to ignore a potential error in production.